cmd/pitr/collector/collector.go

```
@@ -205,54 +222,77 @@ func (c *Collector) manageBinlog(binlog string) (err error) {
+	err = os.Setenv("MYSQL_PWD", os.Getenv("PXC_PASS"))
+	if err != nil {
+		return errors.Wrap(err, "set mysql pwd env var")
+	}
+
+	cmd := exec.Command("mysqlbinlog", "-R", "--raw", "-h"+c.db.GetHost(), "-u"+c.pxcUser, "--result-file="+tmpDir, binlog.Name)
+
+	errOut, err := cmd.StderrPipe()
+	if err != nil {
+		return errors.Wrap(err, "get mysqlbinlog stderr pipe")
+	}
+
+	file, err := os.OpenFile(tmpDir+binlog.Name, os.O_RDWR, os.ModeNamedPipe)
 	if err != nil {
 		return errors.Wrap(err, "open named pipe file error")
 	}
+
 	defer func() {
 		errC := file.Close()
 		if errC != nil {
 			err = mergeErrors(err, errors.Wrapf(errC, "close tmp file for %s", binlog))
 			return
 		}
-		errR := os.Remove(tmpDir + binlog)
+		errR := os.Remove(tmpDir + binlog.Name)
 		if errR != nil {
 			err = mergeErrors(err, errors.Wrapf(errR, "remove tmp file for %s", binlog))
 			return
 		}
 	}()
-	err = os.Setenv("MYSQL_PWD", c.pxcPass)
-	if err != nil {
-		return errors.Wrap(err, "set mysql pwd env var")
-	}
 
-	cmd := exec.Command("mysqlbinlog", "-R", "--raw", "-h"+c.db.GetHost(), "-u"+c.pxcUser, "--result-file="+tmpDir, binlog)
-
-	errOut, err := cmd.StderrPipe()
-	if err != nil {
-		return errors.Wrap(err, "get mysqlbinlog stderr pipe")
+	pipeBuf := &bytes.Buffer{}
+	pr := pipeReader{
+		f:   file,
+		buf: pipeBuf,
 	}
+	go pr.ReadToBuf()
 
 	err = cmd.Start()
 	if err != nil {
 		return errors.Wrap(err, "run mysqlbinlog command")
 	}
 
-	data := reader{file}
-	err = c.storage.PutObject(binlogName, &data)
+	for {
+		time.Sleep(10 * time.Millisecond)
+		if pr.notEmpty {
+			break
+		}
+		stdErr, err := ioutil.ReadAll(errOut)
+		if err != nil {
+			return errors.Wrap(err, "read mysqlbinlog error output")
+		}
+		if len(stdErr) != 0 {
+			return errors.Errorf("mysqlbinlog: %s", stdErr)
+		}
+	}
+
+	err = c.storage.PutObject(binlogName, pipeBuf, -1)
 	if err != nil {
 		return errors.Wrapf(err, "put %s object", binlog)
 	}
+
```

Look at line 260
```
cmd.Wait()
```
Error is not checked

---

https://github.com/percona/percona-xtradb-cluster-operator/pull/723/commits/7476cb110a90a023d52b522c6ab1014e94ad895d